File: frontend/src/types/rappel.ts
// Types correspondant au schéma Pydantic AlertesRappelsOut
// (backend/app/schemas/rappel.py).

export interface EcritureNonSaisie {
  id: string;
  entreprise_id: string;
  entreprise_nom: string;
  document_id: string;
  nom_fichier_document: string;
  tiers: string | null;
  numero_piece: string | null;
  date_piece: string | null;
  montant_ttc: string;
}

export interface EntrepriseEnRetard {
  entreprise_id: string;
  entreprise_nom: string;
  dernier_upload: string;
  delai_reference_jours: number;
  jours_depuis_dernier_upload: number;
  jours_de_retard: number;
}

export interface AlertesRappels {
  non_saisies: EcritureNonSaisie[];
  entreprises_en_retard: EntrepriseEnRetard[];
}

File: backend/app/schemas/rappel.py
"""
app/schemas/rappel.py

Schémas Pydantic pour les alertes automatiques de la page Rappels &
Tâches. Contrairement à Tache (stockée en base, créée manuellement),
ces alertes sont calculées à la volée à chaque appel de l'API --
rien n'est stocké ici, exactement comme pour le Registre comptable.
"""
import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class EcritureNonSaisieOut(BaseModel):
    """Une écriture validée mais pas encore ressaisie dans Topaze."""
    id: uuid.UUID
    entreprise_id: uuid.UUID
    entreprise_nom: str
    document_id: uuid.UUID
    nom_fichier_document: str
    tiers: str | None
    numero_piece: str | None
    date_piece: date | None
    montant_ttc: Decimal


class EntrepriseRetardOut(BaseModel):
    """
    Une entreprise dont le délai habituel entre 2 envois de documents
    est dépassé. Le "délai de référence" est calculé une seule fois par
    entreprise : l'écart entre son 1er et son 2e document jamais uploadé
    (règle métier demandée par l'utilisateur).
    """
    entreprise_id: uuid.UUID
    entreprise_nom: str
    dernier_upload: datetime
    delai_reference_jours: float
    jours_depuis_dernier_upload: float
    jours_de_retard: float


class AlertesRappelsOut(BaseModel):
    """Regroupe les 2 listes d'alertes automatiques en un seul appel API."""
    non_saisies: list[EcritureNonSaisieOut]
    entreprises_en_retard: list[EntrepriseRetardOut]